x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_logic.test.ts

```diff
@@ -15,79 +15,126 @@ import { FlashMessagesLogic, mountFlashMessagesLogic } from './flash_messages_lo
-    resetContext({});
   });
 
   it('has default values', () => {
     mount();
     expect(FlashMessagesLogic.values).toEqual({
       messages: [],
       queuedMessages: [],
+      toastMessages: [],
       historyListener: expect.any(Function),
     });
   });
 
-  describe('setFlashMessages()', () => {
-    it('sets an array of messages', () => {
-      const messages: IFlashMessage[] = [
-        { type: 'success', message: 'Hello world!!' },
-        { type: 'error', message: 'Whoa nelly!', description: 'Uh oh' },
-        { type: 'info', message: 'Everything is fine, nothing is ruined' },
-      ];
-
+  describe('messages', () => {
+    beforeAll(() => {
       mount();
-      FlashMessagesLogic.actions.setFlashMessages(messages);
-
-      expect(FlashMessagesLogic.values.messages).toEqual(messages);
     });
 
-    it('automatically converts to an array if a single message obj is passed in', () => {
-      const message = { type: 'success', message: 'I turn into an array!' } as IFlashMessage;
+    describe('setFlashMessages', () => {
+      it('sets an array of messages', () => {
+        const messages: IFlashMessage[] = [
+          { type: 'success', message: 'Hello world!!' },
+          { type: 'error', message: 'Whoa nelly!', description: 'Uh oh' },
+          { type: 'info', message: 'Everything is fine, nothing is ruined' },
+        ];
 
-      mount();
-      FlashMessagesLogic.actions.setFlashMessages(message);
+        FlashMessagesLogic.actions.setFlashMessages(messages);
+
+        expect(FlashMessagesLogic.values.messages).toEqual(messages);
+      });
+
+      it('automatically converts to an array if a single message obj is passed in', () => {
+        const message = { type: 'success', message: 'I turn into an array!' } as IFlashMessage;
+
+        FlashMessagesLogic.actions.setFlashMessages(message);
 
-      expect(FlashMessagesLogic.values.messages).toEqual([message]);
+        expect(FlashMessagesLogic.values.messages).toEqual([message]);
+      });
     });
-  });
 
-  describe('clearFlashMessages()', () => {
-    it('sets messages back to an empty array', () => {
-      mount();
-      FlashMessagesLogic.actions.setFlashMessages('test' as any);
-      FlashMessagesLogic.actions.clearFlashMessages();
+    describe('clearFlashMessages', () => {
+      it('resets messages back to an empty array', () => {
+        FlashMessagesLogic.actions.clearFlashMessages();
 
-      expect(FlashMessagesLogic.values.messages).toEqual([]);
+        expect(FlashMessagesLogic.values.messages).toEqual([]);
+      });
     });
   });
 
-  describe('setQueuedMessages()', () => {
-    it('sets an array of messages', () => {
-      const queuedMessage: IFlashMessage = { type: 'error', message: 'You deleted a thing' };
-
+  describe('queuedMessages', () => {
+    beforeAll(() => {
       mount();
-      FlashMessagesLogic.actions.setQueuedMessages(queuedMessage);
+    });
+
+    describe('setQueuedMessages', () => {
```

Comment:
I added describe reducer blocks around these actions, so most of these changes are just nesting - I recommend viewing the diff with whitespace changes hidden